### PR TITLE
Fixes #2761 change transport direction

### LINKED ISSUE #2761: Changing transpor direction in the expression is not applied in the simulation
1. Add a transporter protein with direction "Efflux"
2. Add a transport process using the new transporter protein to a molecule
3. Create a simulation and send to MoBi:

image

4. In the Expression Profile, change the direction of the transporter protein to "Influx". Simulation becomes "Red" and the comparison with building block correctly shows the difference in direction:
image

5. Right click on individual BB in the simulation and select "Update from BB"
6. Send the updated simulation to MoBi -> Direction of the transport is still "Efflux":

image

### PR DESCRIPTION
**Pull request by msevestre**
Fixes #2761

# Description

Probem was that it was seen as a non structural change (e.g. changing the expression profile structure). now I am also checking that all used expression profile have the same structure version

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [ ] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs [hunk 1]
```diff
@@ -140,7 +142,33 @@ public bool QuickUpdatePossibleFor(IPKSimBuildingBlock templateBuildingBlock, Us
          if (!BuildingBlockSupportsQuickUpdate(templateBuildingBlock))
             return false;
 
-         return templateBuildingBlock.StructureVersion == usedBuildingBlock.StructureVersion;
+         //not the same structure, easy return
+         var sameStructureVersion = templateBuildingBlock.StructureVersion == usedBuildingBlock.StructureVersion;
+         if (!sameStructureVersion)
+            return false;
+
+         //For individual, there is a special handling required as we need to also check that the used expression profile have hte same structure version
```
Comment (Yuri05) [inline src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:150]: typo: "hte" => the

Comment (Yuri05) [inline src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:150]: Do we need a special handling for _populations_ as well?

File: src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs [hunk 2]
```diff
@@ -140,7 +142,33 @@ public bool QuickUpdatePossibleFor(IPKSimBuildingBlock templateBuildingBlock, Us
...
+         //It has to be available by construction
+         var usedIndividual = usedBuildingBlock.BuildingBlock as Individual;
+         //but we return false just in case :)
+         if (usedIndividual == null)
+            return false;
+
+         //let's compare the expression profile in each individuals and see if they are comparable
+         //not the same length of expression profile, structural change
+         if (usedIndividual.AllExpressionProfiles().Count != individualTemplate.AllExpressionProfiles().Count)
+            return false;
```
Comment (Yuri05) [inline src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:163]: Is it actually possible, that `sameStructureVersion` in the code above is true, but we have a different number of expression profiles?

**Comment by Yuri05** [comment]
If we use a PR template, we should also tick some options when creating a PR 😉

File: src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs (hunk 1 above)
Comment (msevestre) [reply src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:150]: nope. We dont allow commit or update for population

File: src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs (hunk 2 above)
Comment (msevestre) [reply src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:163]: I don't think so. you are right this check can be removed.

**Comment by msevestre** [comment]
Agreed. I forgot to check. Still have to get used to doing this here

File: src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs (hunk 1 above)
Comment (Yuri05) [reply src/PKSim.Core/Services/SimulationBuildingBlockUpdater.cs:150]: ok

**Pull request merged**